### PR TITLE
[vividus-plugin-azure-functions] Introduce Azure Functions support

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/plugins/pages/azure.adoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+= What is Azure
+
+Microsoft https://azure.microsoft.com/[Azure] is a cloud computing service created by Microsoft for building, testing, deploying,
+and managing applications and services through Microsoft-managed data centers.
+
+Vividus provides set of plugin to interact with the services.
```
Comment (valfirst) [inline docs/modules/plugins/pages/azure.adoc:6]: `plugins`

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
+= Azure functions Plugin
+
+The plugin provides functionality to interact with https://azure.microsoft.com/en-us/services/functions/.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:3]: `https://azure.microsoft.com/en-us/services/functions/[Azure Functions]`

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+[source,gradle,subs="attributes+"]
+----
+implementation(group: 'org.vividus', name: 'vividus-plugin-azure-functions', version: '{current-version}')
+----
+
+== Configuration
+
+=== Authentication
+
+By default, plugin's authentication depends on correct configure of following environment variables.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:17]: `plugin's authentication depends on correct configure of following environment variables` ->
`authentication process relies on the configuration of the following environment variables`

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+
+. `AZURE_CLIENT_ID` - for Azure client ID
+. `AZURE_CLIENT_SECRET` - for client secret
+. `AZURE_CLIENT_CERTIFICATE_PATH` - for client certificate
+. `AZURE_TENANT_ID` - for Azure tenant ID
+. `AZURE_SUBSCRIPTION_ID` - Azure subscription ID
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:24]: Do all of them are needed? Or maybe we can define groups of variables specifying different authentication approaches

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+
+[TIP]
+
+One should set either `AZURE_CLIENT_SECRET` or `AZURE_CLIENT_CERTIFICATE_PATH`
+
+
+See the official https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/identity/azure-identity["Azure identity"] guide to get more details.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:32]: `https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/identity/azure-identity#environment-variables["Azure identity"]`

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+== Steps
+
+=== Function triggering
+
+Triggers a function
+
+[source,gherkin]
+----
+When I trigger azure `$functionApp` function app `$function` function in resource group `$resourceGroup` with payload:$payload and save response into `$scopes` variable `$variableNames`
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:49]:
```
When I trigger azure `$functionApp` function app with `$function` function name ...
``` ?

File: vividus-plugin-azure-functions/src/main/java/org/vividus/azure/functions/service/FunctionService.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                                .withPolicy(responseCapturingHttpPipelinePolicy)
+                                .authenticate(credential, azureProfile);
+    }
+
+    public List<Map<String, Object>> triggerFunction(String resourceGroup, String appName, String functionName,
+            Object payload)
+    {
+        ResponseCapturingHttpPipelinePolicy responseCapturingHttpPipelinePolicy =
+                new ResponseCapturingHttpPipelinePolicy();
+        createManager(responseCapturingHttpPipelinePolicy)
```
Comment (valfirst) [inline vividus-plugin-azure-functions/src/main/java/org/vividus/azure/functions/service/FunctionService.java:52]: is app service manager thread safe? does it make sense to make it class field?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+[source,gherkin]
+----
+When I trigger funtion `$functionName` from function app `$functionAppName` in resource group `$resourceGroup` with payload:$payload and save response into `$scopes` variable `$variableNames`
+----
+
+* `$functionName` - the name of the function to trigger
+* `$functionAppName` - the name of the funciton app
+* `$resourceGroup` - the resource group function relates to
+* `$payload` - The JSON payload to send to a function(could be empty)
```
Comment (uarlouski) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:53]: nction (could be empty)

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+Triggers a function
+
+[source,gherkin]
+----
+When I trigger funtion `$functionName` from function app `$functionAppName` in resource group `$resourceGroup` with payload:$payload and save response into `$scopes` variable `$variableNames`
+----
+
+* `$functionName` - the name of the function to trigger
+* `$functionAppName` - the name of the funciton app
```
Comment (uarlouski) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:51]: funciton :)

File: docs/modules/plugins/pages/plugin-azure-functions.adoc [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+----
+When I trigger funtion `$functionName` from function app `$functionAppName` in resource group `$resourceGroup` with payload:$payload and save response into `$scopes` variable `$variableNames`
+----
+
+* `$functionName` - the name of the function to trigger
+* `$functionAppName` - the name of the function app
+* `$resourceGroup` - the resource group function relates to
+* `$payload` - The JSON payload to send to a function(could be empty)
+* `$scopes` - xref:parameters:variable-scope.adoc[The comma-separated set of the variables scopes].
+* `$variableName` - The variable name to store results in JSON format.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-azure-functions.adoc:49]: please document what is saved into the variable (statusCode, body, etc.)

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-azure-functions/src/main/java/org/vividus/azure/functions/steps/FunctionSteps.java [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        this.jsonUtils = jsonUtils;
+        this.functionsService = functionsService;
+        this.bddVariableContext = bddVariableContext;
+    }
+
+    /**
+     * Triggers Azure function by its name. And saves responses from the service into variable with
+     *  a provided scope and name
+     * @param functionApp   the name of Azure Function App. The value can be retrieved by looking
```
Comment (abudevich) [inline vividus-plugin-azure-functions/src/main/java/org/vividus/azure/functions/steps/FunctionSteps.java:48]: Capital letter for description of parameters ? (as https://github.com/vividus-framework/vividus/blob/0943965b8db6c321acf89223946ff1b3a42af7fb/vividus-plugin-azure-functions/src/main/java/org/vividus/azure/functions/steps/FunctionSteps.java#L56)

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**